lib/child-writer/writer-proxy.js

```diff
@@ -157,6 +158,13 @@ return isElevated().then((elevated) => {
       message: 'User did not grant permission.'
     }, () => {
       process.exit(EXIT_CODES.CANCELLED);
+    }).catch({
+      message: 'No polkit authentication agent found.'
```

Comment:
I can't think of anything apart from checking `sudo-prompt` each time we upgrade it, to make sure we have the right error messages :( I agree that this means things can break out of the sudden if `sudo-prompt` changes the error messages, but looks like the best think we can do at the moment.